apps/dc_tools/odc/apps/dc_tools/utils.py

```diff
@@ -156,52 +166,100 @@ def index_update_dataset(
     update=False,
     update_if_exists=False,
     allow_unsafe=False,
-):
-    if uri is not None:
-        # Make sure we can create a dataset first
+    archive_less_mature: Optional[Union[bool, Iterable[str]]]=None
+) -> int:
+    if uri is None:
+        raise IndexingException("Failed to get URI from metadata doc")
+    # Make sure we can create a dataset first
+    try:
+        ds, err = doc2ds(metadata, uri)
+    except ValueError as e:
+        raise IndexingException(
+            f"Exception thrown when trying to create dataset: '{e}'\n The URI was {uri}"
+        )
+    if ds is None:
+        raise IndexingException(
+            f"Failed to create dataset with error {err}\n The URI was {uri}"
+        )
+
+    if archive_less_mature:
+        if archive_less_mature == True:
+            # if set explicitly to True, default to [region_code, time]
+            archive_less_mature = ["region_code", "time"]
```

Comment:
alright, now I understand what `archive_less_mature` kinda changed between `bool` and `None` here and there.

Reply:
Yes, the idea is you could pass in a different list of columns to group by if you really wanted.

`None` and `False` mean don't do maturity checking and `True` means do maturity checking, grouping by `region_code` and `time`.   The CLIs only use True and False, but it is extensible in future if we need to group by something else.

Reply:
I'll try and make the comments a bit clearer.